test/ci-build.sh

```diff
@@ -78,11 +88,18 @@ sanitized_build()
     # bug, cf. https://groups.google.com/forum/#!topic/mesonbuild/tgEdAXIIdC4
     meson configure -D b_lundef=false
 
-    meson configure
+    # additional options
+    meson configure "$@"
+
+    meson configure --no-pager
```

Comment:
We just had an interesting issue on my project, `meson configure` is probably not doing what you expect it to here.

`meson setup` will apply any config options and setup the build, `meson configure` will save configuration options but not setup the build.  The only reason this code works is that the subsequent `ninja` call will check if the options file is newer than the output of setup, and if so it'll re-run `meson setup`, but this depends on file age checks working which itself depends on timing, how quick the script runs and the backing filesystem etc.

After the last configure call and before calling `ninja` you need to add a explicit call to `meson setup --reconfigure "${SRC_DIR"` or similar.

Reply:
The last "meson configure --no-pager" is there to see the resulting options. 
See the current tests - the entire branch is here because tests out of the sudden started to stall. I went back to 3.16.2 and tested that - same stall. 
Anyway, from the output I actually exactly get what I'm expecting.

Reply:
Yes, I agree.  Meson will have saved the configuration options correctly and will show them to you, but it won't have used them to re-generate the ninja config file, if you look at the output the first line from Ninja will be something like "regenerating config" which is ninja detecting the meson config is newer than the meson config and running "meson reconfigure" itself - but this is explicitly racey as it depends on file ages being different.

Reply:
Thanks, I had thought about it again and had just added it. Thanks for your feedback. 

Also added backtraces in the signal handler, as I can't figure from github output why tests are failing. I see that `test_write_cache.c` fails when mnt_dir is missing, but I ensured that it is there.

Reply:
Thank you.

Consider the output at https://github.com/libfuse/libfuse/actions/runs/8357627990/job/22877175705 vs https://github.com/libfuse/libfuse/actions/runs/8357528585/job/22876857707
You can see the reconfigure happening both times but in the second case it's triggered by the ninja command itself, and this configure does need to happen for the options to be applied.

This is a thorny aspect of the meson interface, we should add the extra reconfigure line to https://github.com/libfuse/libfuse/blob/master/README.md?plain=1#L76 as well IMO.

Reply:
Done.